core/src/test/java/com/datastax/oss/driver/internal/core/cql/DirectThreadEventExecutor.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.cql;
+
+import com.datastax.oss.driver.shaded.guava.common.util.concurrent.MoreExecutors;
+import io.netty.util.concurrent.SingleThreadEventExecutor;
+
+public class DirectThreadEventExecutor extends SingleThreadEventExecutor {
+
+  public static final DirectThreadEventExecutor INSTANCE = new DirectThreadEventExecutor();
+
+  private DirectThreadEventExecutor() {
+    super(null, MoreExecutors.directExecutor(), true);
+  }
+
+  @Override
+  protected void run() {
+    // copied from DefaultEventExecutor
+    for (; ; ) {
+      Runnable task = takeTask();
```

Comment:
shouldn't we add some kind of delay between the invocation of this `takeTask()`?
Similar to `GlobalEventExecutor`:
`taskQueue.poll(delayNanos, TimeUnit.NANOSECONDS);` ?

Reply:
I don't see this instruction in `GlobalEventExecutor.run()`.

Reply:
Ok, you right. This is already within `takeTask()` so we should be ok